modules/apps/object/object-service/src/main/java/com/liferay/object/internal/action/executor/FunctionObjectActionExecutorImpl.java

```diff
@@ -62,32 +66,30 @@ public void execute(
 	}
 
 	@Override
-	public String getKey() {
-		return _key;
+	public long getAllowedCompanyId() {
+		return _companyId;
 	}
 
 	@Override
-	public boolean isAllowedCompany(long companyId) {
-		if (_companyId == companyId) {
-			return true;
-		}
-
-		return false;
+	public List<String> getAllowedObjectDefinitionNames() {
+		return _allowedObjectDefinitionNames;
 	}
 
 	@Override
-	public boolean isAllowedObjectDefinition(String objectDefinitionName) {
-		if (_allowedObjectDefinitionNames.isEmpty()) {
-			return true;
-		}
-
-		return _allowedObjectDefinitionNames.contains(objectDefinitionName);
+	public String getKey() {
+		return _key;
 	}
 
 	@Activate
 	protected void activate(Map<String, Object> properties) throws Exception {
 		_allowedObjectDefinitionNames = StringUtil.asList(
 			properties.get("allowedObjectDefinitionNames"));
+
+		if (_allowedObjectDefinitionNames.isEmpty()) {
```

Comment:
Hey @MarinhoFeliphe , do we really need to add the "ALL_OBJECT_DEFINITIONS" in allowedObjectDefinitionNames when it's empty? To simplify I would keep it empty and assume that when it's empty there is no configuration to specific objects, so it should work for all objects.

Reply:
Makes sense to me, thanks Camacho!